lib/firewall/firewall.cc

```diff
@@ -748,3 +753,45 @@ void firewall_remove_udpipv6_remote_endpoint(uint8_t *remoteAddress,
 		  IPProtocolNumber::UDP, *copy, localPort, remotePort);
 	}
 }
+
+bool ethernet_driver_start(std::atomic<uint32_t> *state)
+{
+	if (TCPIPRestartState == nullptr)
+	{
+		if (!CHERI::check_pointer<CHERI::PermissionSet{
+		      CHERI::Permission::Load, CHERI::Permission::Global}>(
+		      state, sizeof(*state)))
+		{
+			Debug::log("Invalid TCP/IP state pointer {}", state);
+			return false;
+		}
+		TCPIPRestartState = state;
+	}
+	if (TCPIPRestartState->load() != 0)
+	{
+		// This is a restart, no need to actually reset the driver.
+		// Instead, just remove all firewall entries.
+		Debug::log("Network stack restart: clearing all entries.");
+		EndpointsTable<IPv6Address>::instance().clear(IPProtocolNumber::UDP);
+		EndpointsTable<IPv6Address>::instance().clear(IPProtocolNumber::TCP);
```

Comment:
Right - I cannot really fix that for now though because your PR which enables to compile out the IPv6 stuff has not been merged yet: https://github.com/CHERIoT-Platform/network-stack/pull/26

I suggest to leave it as it is for now and fix it when we rebase, either here (if you PR gets merged first), or in your PR (if this one gets merged first).